src/RequestProcessor.cs

```diff
@@ -355,6 +356,14 @@ internal StreamingMessage ProcessFunctionEnvironmentReloadRequest(StreamingMessa
 
             var rpcLogger = new RpcLogger(_msgStream);
             rpcLogger.SetContext(request.RequestId, null);
+
+            if (environmentReloadRequest.FunctionAppDirectory != null)
+            {
```

Comment:
Please add unit test. Also, open issue here: https://github.com/Azure/azure-functions-integration-tests/issues/ - to track E2E test that verifies app directory after specialization.

Reply:
Thank you. Added: https://github.com/Azure/azure-functions-integration-tests/issues/10